### PR TITLE
[DONE] Add indication whether controller callback was successfully disconnected

### PR DESCRIPTION
**Pull request by Swiftb0y**
While Mapping the NS6II, I encountered a problem where I had to check if the connection is already disconnected. 
The added code will set `connection.isConnected` to true/false accordingly. It will also return whether `connection.disconnect()` was successful.

TODO:
- [x] Add tests
- [x] Add Wiki Documentation

### CODE REVIEW

**Comment by Be-ing** [comment]
Nice idea. Would it be better to add a `conn.isActive` or `conn.isConnected` property instead of adding another meaning to the `id` property?

**Comment by Swiftb0y** [comment]
It would make more sense yes. I just thought using the id would be fine since it has no real purpose in mappings right now. Also `isConnected` would IMO hint that it was possible to reconnect a disconnected "callback object". Regardless, you are probably right that a dedicated property would be the better choice.

File: src/controllers/controllerengine.cpp [hunk 1]
```diff
@@ -764,19 +764,22 @@ void ScriptConnection::executeCallback(double value) const {
...
 
-    coScript->removeScriptConnection(connection);
+    return coScript->removeScriptConnection(connection);
 }
 
-void ScriptConnectionInvokableWrapper::disconnect() {
-    m_scriptConnection.controllerEngine->removeScriptConnection(m_scriptConnection);
+bool ScriptConnectionInvokableWrapper::disconnect() {
+    // if the removeScriptConnection succeeded, the connection has been successfully disconnected
+	bool success = m_scriptConnection.controllerEngine->removeScriptConnection(m_scriptConnection);
```
Comment (Be-ing) [inline src/controllers/controllerengine.cpp]:
```suggestion
    bool success = m_scriptConnection.controllerEngine->removeScriptConnection(m_scriptConnection);
```
replace tab with space

**Comment by Be-ing** [comment]
Looks good to me other than the tab character noted above. Thanks for working on this and documenting it, I think it will be useful for debugging controller scripts. I edited the documentation you added to the wiki to note this will be new in Mixxx 2.3.

**Pull request merged**